test/parallel/test-icu-env.js

```diff
@@ -122,22 +122,22 @@ if (isMockable) {
   assert.deepStrictEqual(
     locales.map((LANG) => runEnvOutside({ LANG, TZ: 'Europe/Zurich' }, 'new Date(333333333333).toLocaleString()')),
     [
-      '7/25/1980, 1:35:33 AM',
+      '7/25/1980, 1:35:33 AM',
```

Comment:
I suspect this will break the sharedlib ICU builds which will be using the previous format?

Reply:
That's what happened. Should I make two branches depending on the ICU version?

Reply:
Yes. I think technically the tests could be branched on the cldr version but in practice the ICU version is probably sufficient? At least we don't/haven't updated cldr independently of ICU version bumps up until now.

cc @srl295

Reply:
cldr doesn't usually update separate from icu

Reply:
Huh, this is actually a _new_ test - https://github.com/nodejs/node/pull/42683  - just a few months old:

I have concerns with the premise of this test:

> This test gives an example of observable results and tests if this behaviour is kept.

this behaviour is guaranteed NOT to be stable across releases.  I'd drop the test, or make it keyed on an exact CLDR version.

Reply:
/cc @LiviaMedeiros

Reply:
I'll add comments on the #42440 issue also, there's some discussion there.

Reply:
> this is always the case if you link with a backlevel ICU. I assume sharedlib = local system icu?

Yes, sort of. We build several versions of ICU as a shared library in a container and then build and dynamically link Node.js against the minimum ICU version that release line supports (the highest out of V8's minimum ICU version and the version of ICU Node.js' GA release (e.g. 18.0.0) went out with). We don't officially release binaries for this but do it to catch issues that may come up for downstream packagers (e.g. Linux distros) who build against the system ICU.

Reply:
added

Reply:
@richardlau ^ that seems to be an argument for making these tests specific to the ICU+CLDR version… otherwise, skipping them.

Reply:
This test aims only on builds with `full-icu`. Both [`system-icu`](https://github.com/LiviaMedeiros/node/blob/6686d9000b64a8cbfa2077ea0e30e253c35e053a/test/parallel/test-icu-env.js#L7-L8) and [`small-icu`](https://github.com/LiviaMedeiros/node/blob/6686d9000b64a8cbfa2077ea0e30e253c35e053a/test/parallel/test-icu-env.js#L21-L22) are skipped.
The change makes sense to me within any semverity.

Reply:
@LiviaMedeiros  I think it should be skipped unless the icu, cldr, tz, and unicode versions all match. Because a ful icu build can be a different version than the in-repo one.

Reply:
> The change makes sense to me within any semverity.

Which change? Given that you added the test, do you think we need to consider the ICU 72.1 update a semver-major change?

Reply:
> Which change?

The reviewed part, and the rest of changes in this file in this PR: https://github.com/nodejs/node/pull/45068/files#diff-a6095e09d738de6d1c78d336fac497210d041350d1ece89ef04491ff22bbf4b4L125-R135

> Given that you added the test, do you think we need to consider the ICU 72.1 update a semver-major change?

No, the opposite: I consider changes in this file in this PR to be `semver-patch`, since it only slightly changes formatting in a few locales.

However, *if* the test would fail for a more serious reason (for example, adding a global cache for ICU that overrides environment variables even in freshly spawned process), I think it would be safer to not backport it.
The purpose of this test is to detect *if* something like that happens.

Reply:
> it might lead to test failures in other places as well anyway

That's an argument for not having such tests.

I'm trying to understand what the intent of testing is… is it to verify that ICU is integrated? Or that ICU's data never changes?

Because the data changes frequently. Some examples from the past few months:   Sierra Leone revaluates its currency.  Croatia switches to the Euro.  Turkey is now Türkiye.  Date formats change due to linguistic and geopolitical requests.  Kazakhistan switches to Latin script. Etc….  the output here is not generally suitable for regression tests that are expected to pass in the future without modification.

Reply:
CLDR 42 was released yesterday. [here](https://unicode-org.github.io/cldr-staging/charts/42/delta/index.html) is the delta chart with _some_ of the data changes.  To give an idea of the magnitude of what changes from release to release.